data/data/azurestack/vnet/nsg.tf

```diff
@@ -0,0 +1,19 @@
+resource "azurestack_network_security_group" "cluster" {
+  name                = "${var.cluster_id}-nsg"
+  location            = var.azure_region
+  resource_group_name = data.azurestack_resource_group.main.name
+}
+resource "azurestack_network_security_rule" "apiserver_in" {
+  name                        = "apiserver_in"
+  priority                    = 101
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = "6443"
+  source_address_prefix       = "*"
+  destination_address_prefix  = "*"
+  resource_group_name         = data.azurestack_resource_group.main.name
+  network_security_group_name = azurestack_network_security_group.cluster.name
+  description                 = local.description
+}
```

Comment:
Matthew's work also included the `azurestack_subnet_network_security_group_association` which I added to the terraform provider. I could add all of these later too if you like.